src/main/ruby/core/enumerable.rb

```diff
@@ -112,6 +112,7 @@ def count(item = undefined)
     seq = 0
     if !undefined.equal?(item)
       each { |o| seq += 1 if item == o }
+      warn 'warning: given block not used' if block_given?
```

Comment:
I wonder about fixing this upstream. Should I change it to use the intuitive behavior or do as CRuby does for now?